### PR TITLE
'rhc app show <app> --gears quota' use 'quota' not 'du'

### PR DESCRIPTION
**Pull request by sallyom**
Bug 1109935
Bugzilla https://bugzilla.redhat.com/show_bug.cgi?id=1109935

Users expect 'rhc app show <app> --gears quota' to display output of the system 'quota' cmd.
Currently, the rhc cmd uses 'du -s'.  This PR replaces 'du' with 'quota'

### CODE REVIEW

File: lib/rhc/commands/app.rb [hunk 1]
```diff
@@ -425,7 +425,8 @@ def show(app_name)
         case options.gears
         when 'quota'
           opts = {:as => :gear, :split_cells_on => /\s*\t/, :header => ['Gear', 'Cartridges', 'Used', 'Limit'], :align => [nil, nil, :right, :right]}
-          table_from_gears('echo "$(du --block-size=1 -s 2>/dev/null | cut -f 1)"', groups, opts) do |gear, data, group|
+          # quota 'blocks' are actually kb, so have to multiply by 1024, see 'man quota'
+          table_from_gears('echo "$(quota | egrep -v \'quotas|Filesystem\' | awk \'{ print $2 * 1024 }\')"', groups, opts) do |gear, data, group|
```
Comment (Miciah) [inline lib/rhc/commands/app.rb]: `echo "$(foo)"` can be rewritten as `foo`.

Comment (Miciah) [inline lib/rhc/commands/app.rb]: I'd rather not make assumptions about the blocksize, but I cannot find a really good way to look it up.  The following works but is kind of excessive: `cpp -include sys/mount.h -dM < /dev/null | awk '$2 == "BLOCK_SIZE" { print $3 }'`

Comment (Miciah) [inline lib/rhc/commands/app.rb]: Might be more robust just to skip the first two lines, or even to select the third line specifically: `quota | awk 'NR == 3 { print $2 * 1024 }'`

Comment (sallyom) [inline lib/rhc/commands/app.rb]: @Miciah 
- replaced the egrep with NR = 3 :)
- from command line, no need for 'echo "$(foo)" but this format is the only one that works in this case, I've tried everything! I can investigate further into why, but that's the way the du cmd is currently, maybe for the same unknown reason.
- I agree the blocksize shouldn't be hardcoded.  'cat /usr/include/sys/mount.h | grep BLOCK_SIZE | awk ' NR=1 {print $3}' also works but it's so far been tricky to get this value into the quota cmd.  There must be a way, I'll figure it out (unless you know?)
thanks!

Comment (sallyom) [inline lib/rhc/commands/app.rb]: @Miciah Actually, looking at the man page for quota:
> quota  displays  users' disk usage and limits.  By default only the user quotas are printed. **By default space usage and limits are shown in kbytes (and are named blocks  for  historical  reasons).**

So it actually is always 1024 (kbytes) not BLOCK_SIZE from mount.h  so it seems like hard-coding makes sense?

Comment (Miciah) [inline lib/rhc/commands/app.rb]: Oh, you need to use `echo "$(quota ...)"` because `oo-trap-user` recognises the `quota` command and runs it with `execv` (cf. https://github.com/openshift/origin-server/blob/master/node/misc/bin/oo-trap-user#L310-L335) whereas `oo-trap-user` does not recognise `echo` and instead runs it with `bash -c`, so pipes work (cf. https://github.com/openshift/origin-server/blob/master/node/misc/bin/oo-trap-user#L216-L224).

`command quota | awk 'NR == 3 { print $2 * 1024 }'` would also work, but since we're using `echo "$(foo)"` elsewhere we can stick to that idiom.

Comment (Miciah) [inline lib/rhc/commands/app.rb]: Which manpage are you reading? The `quota(1)` manpages on RHEL67 and RHEL71 don't mention that.  But 1024 is probably safe enough; it's probably always 1024 on any system v2 supports (viz. RHEL6 on x86_64).

File: lib/rhc/commands/app.rb [hunk 2]
```diff
@@ -4,6 +4,7 @@
 require 'rhc/cartridge_helpers'
 require 'rhc/deployment_helpers'
 require 'optparse'
+require 'open3'
```
Comment (Miciah) [inline lib/rhc/commands/app.rb]: Why is this needed?

Comment (sallyom) [inline lib/rhc/commands/app.rb]: @Miciah oops, thanks, removing now!

File: lib/rhc/commands/app.rb (hunk 1 above)
Comment (sallyom) [inline lib/rhc/commands/app.rb]: Mystery solved with the 'echo "$(quota...)" ', thank you!
I'm looking at fedora 23 man quota.  Weird.  It's in the 'DESCRIPTION' section.
I'm fine with keeping it '1024' or I could read it from mount.h, either way.
:)

**Comment by Miciah** [comment]
Looks good.  Please [test]!

**Comment by Miciah** [comment]
Please [merge]!
